conversions/decimal_to_binary_recursion.c

```diff
@@ -0,0 +1,23 @@
+#include <assert.h>
+
+/**
+ * Decimal to binary using recursion algorithm
+ */
```

Comment:
```suggestion
 * For example, if number = 5, the function returns the decimal integer 101.
 * @param number positive integer number to convert
 * @returns integer with digits representing binary value representation of number. 
 */
```
1. the input should be an unsigned integer, because this implementation only considers positive numbers and negative numbers are not accounted for
2. This implementation is a bit confusing with regards to its return. It returns another decimal integer that will quickly overflow. Wouldn't it be right to return a string representation? While it is not wrong to implement this way, just wanted to mention.